src/ccall/src/main.rs

```diff
@@ -46,74 +46,70 @@ fn breed_goats(
     println!("generating goat");
     for x in -2..3 {
         for y in -1..2 {
-            loop {
-                let new_goat = breed(parent1, parent2);
-                let (p, n, tc, f) = new_goat.mesh().buffers();
-                let pos_vec = p
-                    .to_vec()
-                    .chunks(3)
-                    .into_iter()
-                    .map(|x| [x[0] as f32, x[1] as f32, x[2] as f32])
-                    .collect::<Vec<_>>();
-
-                let norm_vec = n
-                    .to_vec()
-                    .chunks(3)
-                    .into_iter()
-                    .map(|x| [x[0] as f32, x[1] as f32, x[2] as f32])
-                    .collect::<Vec<_>>();
-                let text_vec = tc
-                    .to_vec()
-                    .chunks(2)
-                    .into_iter()
-                    .map(|x| [x[0] as f32, x[1] as f32])
-                    .collect::<Vec<_>>();
-                let mut mesh = Mesh::new(PrimitiveTopology::TriangleList);
-
-                mesh.set_attribute(
-                    Mesh::ATTRIBUTE_POSITION,
-                    VertexAttributeValues::Float3(pos_vec),
-                );
-                mesh.set_attribute(
-                    Mesh::ATTRIBUTE_NORMAL,
-                    VertexAttributeValues::Float3(norm_vec),
-                );
-                mesh.set_attribute(
-                    Mesh::ATTRIBUTE_UV_0,
-                    VertexAttributeValues::Float2(text_vec),
-                );
-                let faces_vec = f.to_vec();
-                // below is a hack to work around bad goat generation causing panics with mod_picking
-                if faces_vec.iter().all(|x| *x <= 900_000_000) {
-                    mesh.set_indices(Some(Indices::U32(faces_vec)));
-
-                    let mesh_handle = meshes.add(mesh);
-                    let material_handle = materials.add(StandardMaterial {
-                        albedo: Color::rgb(0.8, 0.7, 0.6),
+            let new_goat = breed(parent1, parent2);
+            let goat_mesh = new_goat.mesh();
```

Comment:
creating variable here causes the Mesh object to live long enough.

Reply:
Maybe try https://doc.rust-lang.org/std/marker/struct.PhantomData.html